### PR TITLE
Adds support for classes in Tables

### PR DESCRIPTION
**Pull request by JJ**
Basically so that we can make them sortable refs perl6/doc#1157

### CODE REVIEW

File: lib/Pod/To/HTML.pm [hunk 1]
```diff
@@ -387,7 +386,7 @@ multi sub node2html(Pod::Block::Para $node) {
 
 multi sub node2html(Pod::Block::Table $node) {
     Debug { note colored("Table node2html called for ", "bold") ~ $node.gist };
-    my @r = '<table class="pod-table">';
+    my @r = $node.config<class>??'<table class="pod-table '~$node.config<class>~'">'!!'<table class="pod-table">';
```
Comment (AlexDaniel) [inline lib/Pod/To/HTML.pm]: Hm… not sure if I like the fact that you can attempt injection attacks through this… Do we have any other places where this is possible?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
-.precomp
+**/.precomp/
+
```
Comment (zoffixznet) [inline .gitignore]: This is the third time you make this change. I think your fork is not synced up with all the latest changes when you start working on your PRs.

`.precomp` alone already blocks `.precomp` name in *all* subdirectories. Your change just breaks that behaviour on systems where `**/` doesn't get expanded to mean all subdirs.

**Comment by zoffixznet** [comment]
Without looking at the code, I'd say this is just piece-meal over-engineering and shouldn't go in.

- What's so special about tables that they get a settable class, yet no other elements have this option?
- What's so special about `class` attribute that the module provides a means to set it?
- Why is the assumption that "class" pod configuration would definitely contain an HTML class name, considering there's no connection between Pod and HTML formats? The code doesn't even bother properly escaping the data. I see someone already stuck the same `class` config var handling on the `pod` block, but since it generates invalid HTML (`<span>` elements cannot contain block-level elements, which is what most of POD that'd go inside would contain), I imagine not many use that feature.

In other words, we wanted to do X with the docs site, so we modified this module to be able to do that X alone, but the change wasn't evaluated with respect to this module as a standalone piece of software it is. 

------

The `=for Xhtml` blocks already let the users add custom bits of markup, so IMO we should keep the module simple to use, and not offer a billion knobs to tweak every minute piece of markup

------

Looking at the original https://github.com/perl6/doc/issues/1157 , I don't see why this change is even necessary. We already have a class on the tables and even without it, tables are a fairly easy thing to target.

File: .gitignore (hunk 2 above)
Comment (JJ) [reply .gitignore]: That line is interpreted by git, not by the system. I did add it because .precomp did show up in subdirectories.

File: lib/Pod/To/HTML.pm (hunk 1 above)
Comment (JJ) [reply lib/Pod/To/HTML.pm]: I'm not sure how to fix that...

File: .gitignore (hunk 2 above)
Comment (zoffixznet) [reply .gitignore]: Then just keep both of them in. I know the `**/` in `.gitignore` doesn't work on my ubuntu box.

Comment (zoffixznet) [reply .gitignore]: Those are 2.16.3 docs. Not all repos have latest and greatest. I don't see anything in those docs that'd explain why simple `.precomp` pattern doesn't work for you.

```
zoffix@~/R/rakudo (master)$ mkdir src/.precomp
zoffix@~/R/rakudo (master)$ gd
diff --git a/.gitignore b/.gitignore
index 3d1a43d..efb9f5d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -100,7 +100,7 @@ perl6-gdb-m
 perl6-valgrind-m
 *.rakudo.moar
 *.rakudo.jvm
-.precomp
+**/.precomp
 t/03-jvm/*.class
 /zef
 .DS_Store
zoffix@~/R/rakudo (master)$ gs
# On branch master
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#       modified:   .gitignore
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#       .precomp/
#       t/02-rakudo/test-packages/.precomp/
#       t/04-nativecall/.precomp/
#       t/09-moar/.precomp/
#       t/packages/.precomp/
no changes added to commit (use "git add" and/or "git commit -a")
zoffix@~/R/rakudo (master)$ git --version
git version 1.7.9.5
zoffix@~/R/rakudo (master)$
```

**Comment by JJ** [comment]
Both work for me. I can change that back, no problem. I do have git 2.16.2.
Point is, it's valid .gitignore.

**Comment by zoffixznet** [comment]
> Tables are especial because perl6/doc#1157. 

That's just circular reasoning. I could equally counter-argue and say lists are special because I'm using this module on a site where I want to style them as accordions. I'm not doing anything with tables, so clearly we only need to add class-twirling knobs to lists only.

> We don't want to make all tables sortable

As far as this module is concerned, no tables are sortable. There aren't any jQuery plugins included with the module to do the sorting. You're just spilling your work on docs.perl6.org—a specific website—to a 3rd party ecosystem module for converting Pod to HTML. 

> We want to make a minimum amount of change to existing code, not change all sortable tables to HTML.

That's precisely my point: you're expending a minimal amount of effort to add table sorting plugin to *a* website you're working on. No thought is being given on the impact of this addition to all the other uses of this module.

> Any other way of doing that will be greatly appreciated.

Yeah, I can easily add the sorters.

Would you clarify this though: you said "we don't want to make all tables sortable" but the issue's title asks to make *all* tables sortable. Which tables shouldn't be sortable?

**Comment by JJ** [comment]
Hi

2018-04-02 7:44 GMT+02:00 Zoffix Znet <notifications@github.com>:

>
> Any other way of doing that will be greatly appreciated.
>
> Yeah, I can easily add the sorters.
>

Please do that, by all means.

> Would you clarify this though: you said "we don't want to make all tables
> sortable" but the issue's title asks to make *all* tables sortable. Which
> tables shouldn't be sortable?
>
Maybe those that are not on docs.perl6.org. Even in docs.perl6.org, some
tables, specially those generated automatically, don't need to be
user-sortable.
